### PR TITLE
feat(python): serializable python functions in expressions

### PR DESCRIPTION
**Pull request by ritchie46**
A python function passed to `some_expr().map(..)` now is serializable with `pickle`.

### CODE REVIEW

**Comment by ritchie46** [comment]
@alexander-beedie I don't understand the failing parametric test, can you take a look? Something about a module?

**Comment by alexander-beedie** [comment]
> @alexander-beedie I don't understand the failing parametric test, can you take a look? Something about a module?

Hmm... smells odd/transient; I've taken a dump of the log/stack and triggered a re-run of the failed tests to see if it comes back! In the meantime I'll see if it can be reproduced locally.

File: polars/polars-lazy/polars-plan/src/dsl/python_udf.rs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+        Python::with_gil(|py| {
+            let func: PyObject = PyModule::from_code(
+                py,
+                r#"
+def pickle_python_udf(arg):
+    import pickle
+    return pickle.loads(arg)
+"#,
+                "",
```
Comment (ritchie46) [inline polars/polars-lazy/polars-plan/src/dsl/python_udf.rs:50]: @alexander-beedie note that this creates a python module called `""`. Maybe that gets registered in the python process?

Comment (alexander-beedie) [reply polars/polars-lazy/polars-plan/src/dsl/python_udf.rs:50]: Definitely connected; interestingly the `__init__.py` seemed to knock the errors down from 1 to 2, but that may be a coincidence 🤷   Looks like an artefact of tracing the expected error. I can dig in at home in a few hours.

**Review by alexander-beedie — APPROVED** [review]

**Pull request merged**